include/boost/wave/util/flex_string.hpp

```diff
@@ -549,8 +549,7 @@ class AllocatorStringStorage : public A
 
     void* Alloc(size_type sz, const void* p = 0)
     {
-        return A::allocate(1 + (sz - 1) / sizeof(E),
-            static_cast<const char*>(p));
+        return A::allocate(1 + (sz - 1) / sizeof(E));
```

Comment:
This change needs to be protected by a feature macro or similar to retain compatibility with older compilers. The new signature is available starting C++17 only.

Reply:
Better yet, this could be replaced with the equivalent `allocator_traits` incantation, e.g.:
```
    return std::allocator_traits<A>::allocate(1 + (sz - 1) / sizeof(E));
```